v2/internal/testcommon/kube_per_test_context.go

```diff
@@ -602,21 +602,46 @@ func (tc *KubePerTestContext) AsExtensionOwner(obj client.Object) *genruntime.Ar
 	}
 }
 
-func (tc *KubePerTestContext) ExportAsSample(resource runtime.Object) {
+func (tc *KubePerTestContext) ExportAsSample(resource client.Object) {
 	tc.T.Helper()
 
 	filename := fmt.Sprintf("%s.yaml", tc.T.Name())
 	filepath := path.Join(os.TempDir(), filename)
 
-	copy := resource.DeepCopyObject()
-	err := tc.exportAsYAML(copy, filepath)
+	rsrc := resource.DeepCopyObject()
+	tc.cleanSample(rsrc)
+
+	err := tc.exportAsYAML(rsrc, filepath)
 	if err != nil {
 		tc.T.Fatalf("failed to export resource: %s", err)
 	}
 
 	tc.T.Logf("Exported resource to %s", filepath)
 }
 
+func (tc *KubePerTestContext) cleanSample(resource any) {
+
+	if kr, ok := resource.(genruntime.KubernetesResource); ok {
+		// Remove Status
+		emptyStatus := kr.NewEmptyStatus()
+		_ = kr.SetStatus(emptyStatus) // Ignore errors
```

Comment:
do we need empty status here or nil status here? As seen in the sample `status: {}` is present.

Reply:
We actually want no status at all; will modify.